xarray/tests/test_netcdftimeindex.py

```diff
@@ -0,0 +1,301 @@
+import pandas as pd
+import xarray as xr
+
+from . import TestCase, requires_netCDF4
+
+
+class TestISODateParser(TestCase):
+    def test_parse_iso8601(self):
+        from xarray.core.netcdftimeindex import parse_iso8601
+
+        def date_dict(year=None, month=None, day=None,
+                      hour=None, minute=None, second=None):
+            return dict(year=year, month=month, day=day, hour=hour,
+                        minute=minute, second=second)
+
+        for string, expected in [
+            ('1999', date_dict(year='1999')),
+            ('199901', date_dict(year='1999', month='01')),
+            ('1999-01', date_dict(year='1999', month='01')),
+            ('19990101', date_dict(year='1999', month='01', day='01')),
+            ('1999-01-01', date_dict(year='1999', month='01', day='01')),
+            ('19990101T12', date_dict(year='1999', month='01', day='01',
+                                      hour='12')),
+            ('1999-01-01T12', date_dict(year='1999', month='01', day='01',
+                                        hour='12')),
+            ('19990101T1234', date_dict(year='1999', month='01', day='01',
+                                        hour='12', minute='34')),
+            ('1999-01-01T12:34', date_dict(year='1999', month='01', day='01',
+                                           hour='12', minute='34')),
+            ('19990101T123456', date_dict(year='1999', month='01', day='01',
+                                          hour='12', minute='34',
+                                          second='56')),
+            ('1999-01-01T12:34:56', date_dict(year='1999', month='01',
+                                              day='01', hour='12', minute='34',
+                                              second='56')),
+            ('19990101T123456.78', date_dict(year='1999', month='01',
+                                             day='01', hour='12', minute='34',
+                                             second='56.78')),
+            ('1999-01-01T12:34:56.78', date_dict(year='1999', month='01',
+                                                 day='01', hour='12',
+                                                 minute='34', second='56.78'))
+        ]:
+            result = parse_iso8601(string)
+            self.assertEqual(result, expected)
+
+            if '.' not in string:
+                with self.assertRaises(ValueError):
+                    parse_iso8601(string + '3')
+
+
+@requires_netCDF4
+class NetCDFTimeIndexTests(object):
```

Comment:
Yes the tests do run, though I agree I was a bit careless with the names of the classes:
https://travis-ci.org/pydata/xarray/jobs/198709337#L1877

I'll address that as I refactor things to take more advantage of ```pytest```.